src/client/mod.rs

```diff
@@ -300,7 +357,7 @@ impl WarpgrapherClient {
             "WarpGrapherClient read_node -- query: {:#?}, input: {:#?}",
             query, input
         );
-        let result = graphql(self.endpoint.to_owned(), query, input.cloned())?;
+        let result = graphql(self.endpoint.to_owned(), query, input.cloned()).await?;
```

Comment:
If ownership is needed here, better to take ownership in the argument, above, rather than taking a reference.  Let the caller decide whether to clone.